### DISCUSSION TITLE
.env variable outside plugins, nuxt.config.ts, etc.

### DISCUSSION THREAD

**Discussion by nevadavid** [nuxt/nuxt#17988] [Questions]
I have some utils files (/utils) and I would like to access `publicRuntimeConfig` or `process.env` in client side.
Currently, I get `process.env` variables just on server side.
Is it possible to access it somehow on client side?

**Comment by Diizzayy** [comment]
@nevadavid  It is indeed possible, you can access `publicRuntimeConfig` client side

[Runtime Config Docs](https://v3.nuxtjs.org/docs/usage/runtime-config) - You can better familiarize yourself with runtime config by reading the documentation.

`nuxt.config.ts`
```ts
publicRuntimeConfig: {
  secret: process.env.MY_SECRET
},
```

after adding your variable to runtime config, it can be access in your page / component as seen below

```ts
<script lang="ts" setup>
const config = useRuntimeConfig();

onMounted(() => {
  console.log("My Secret", config?.secret);
});
</script>
```
